src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -36,11 +42,57 @@ var Marathon = React.createClass({
       deploymentsFetchState: States.STATE_LOADING,
       fetchState: States.STATE_LOADING,
       modalClass: null,
+      route: null,
       tasksFetchState: States.STATE_LOADING
     };
   },
 
   componentDidMount: function () {
+    var router = this.props.router;
+
+    router.on("route", function (route, params) {
+      this.setState({
+        route: {
+          name: route,
+          params: params
+        }
+      });
+    }.bind(this));
+
+    router.on("route:about", function () {
+      this.modalDestroy();
+      this.setState({
+        modalClass: AboutModalComponent
+      });
+    }.bind(this));
+
+    router.on("route:apps", function (appid) {
+      if (appid) {
```

Comment:
Or maybe just use `var app = this.state.collection.get("/" + appId);` because the collection should be initialised. Or am I missing something?